gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_clusters.h

```diff
@@ -132,6 +132,17 @@ namespace pcl
           */
         void extract (std::vector<pcl::PointIndices> &clusters);
 
+        /** brief Fast data download from the device to host
+         * param source_indices gpu memory buffer from which to download
+         * param buffer_indices indices denoting positions to download
+         * param buffer_size length of a memory buffer
+         * param downloaded_indices memory buffer of the downloaded indices */
+        static void
```

Comment:
I don't think this should be:
* public (strongly)
* class member (meh, it can be but does it have to? It'll become part of the API and ABI and add maintenance overhead)

Reply:
Thank you for your detailed and very instructive comment, @kunaltyagi ! I put the function inside the class only with great trepidation but thought this was the best fit. However, your comments point out well why this was not a good choice. I placed the function thus again inside the `pcl::detail` namespace. Does that address your concerns?